salt/modules/opkg.py

```diff
@@ -370,15 +374,172 @@ def _parse_reported_packages_from_install_output(output):
 
     return reported_pkgs
 
+def _get_total_packages(cmd):
+    '''
+    Runs the given command in test-mode (--noaction) and returns the number of packages
+    that have to be installed/removed/upgraded/downgraded
+    '''
+    test_cmd = copy.deepcopy(cmd)
+    test_cmd.append('--noaction')
+
+    out = _call_opkg(test_cmd)
+    if out['retcode'] != 0:
+        return -1
+
+    output_lines = out['stdout'].split('\n')
+    return sum(_get_operation_from_output_line(line, False) is not None for line in output_lines)
+
+def _get_operation_from_output_line(line, include_download = True):
+    '''
+    Gets the current opkg operation from output
+    Example:
+        'Installing vim (17.0.0) on root'
+    Returns: 'install'
+    '''
+    operations = {
+            'Installing': 'install',
+            'Upgrading': 'upgrade',
+            'Removing': 'remove',
+            'Downgrading': 'downgrade'
+            }
+    if include_download:
+        operations['Downloading'] = 'download'
+
+    line_tokens = line.split()
+    if not line_tokens:
```

Comment:
instead of these 4 lines just write: 
```
return operations.get(line_tokens[0]) if line_tokens else None
```

Note that you don't need to supply None as default value to operations.get  because it's already the default default_value